### PR TITLE
Updated Add Feature and Other Minor Fixes

### PR DESCRIPTION
**Pull request by maypfv**
In this PR, the Add feature has been updated to better suit our app's requirements. 

Main changes include: Refactoring Name to Description of a Task, Required Fields of an Add Command to be Description with d/ prefix only, Display Messages Upon Adding a Task Successfully and Not Successfully is Simplified.

Additional changes were made to some of the Test files related to Add and Edit feature (see Notes).

Notes: Some of the Name to Description refactoring were also done in the Edit feature since it needed the Name of a task.

### CODE REVIEW

**Comment by seantehds** [comment]
I think the code looks good!

**Review by Sebtey — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -85,14 +85,14 @@ Guarantees:
...
-&ensp;&ensp;1a1. System warns that the <u>[command is illegal (UC07)](#UC07-Warn-on-Illegal-Command)</u>.  
+1a. User enters an illegal command.
+&ensp;&ensp;1a1. System warns that the <u>[command is illegal (UC07)](#UC07-Warn-on-Illegal-Command)</u>.
 &ensp;&ensp;1a2. User acknowledges and closes the warning.
 
 Use case ends.
 
 ### UC02: Delete a task
-Actor(s): Project Manager  
+Actor(s): Project Manager
```
Comment (JCSnap) [inline docs/DeveloperGuide.md:95]: I might be wrong but removing the double space might cause `Guarantee` to be at the same line as `Project Manager`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -127,15 +127,15 @@ Use case ends.
...
 &ensp;&ensp;1a2. User acknowledges and closes the warning.
 
 Use case ends.
 
 ### UC04: Marks a task as done
 
-Actor(s): Project Manager  
+Actor(s): Project Manager
 Guarantees:
 * The specified task is successfully marked as complete in the system.
```
Comment (JCSnap) [inline docs/DeveloperGuide.md:140]: same with this

**Review by JCSnap — APPROVED** [review]

**Review by asdfghjkxd — APPROVED** [review]
LGTM

**Review by seantehds — COMMENTED** [review]
LGTM May! Thanks for working on the different documentation as well!

**Pull request merged**